tools/cli/factory.go

```diff
@@ -95,16 +101,65 @@ func (b *clientFactory) SDKClient(c *cli.Context, namespace string) sdkclient.Cl
 	return sdkClient
 }
 
-func (b *clientFactory) createGRPCConnection(hostPort string) *grpc.ClientConn {
+func (b *clientFactory) createGRPCConnection(c *cli.Context) *grpc.ClientConn {
+	hostPort := c.GlobalString(FlagAddress)
 	if hostPort == "" {
 		hostPort = localHostPort
 	}
+	// Ignoring error as we'll fail to dial anyway, and that will produce a meaningful error
+	host, _, _ := net.SplitHostPort(hostPort)
+
+	certPath := c.GlobalString(FlagTLSCertPath)
+	keyPath := c.GlobalString(FlagTLSKeyPath)
+	caPath := c.GlobalString(FlagTLSCaPath)
+
+	grpcSecurityOptions := grpc.WithInsecure()
+	var cert *tls.Certificate
+	var caPool *x509.CertPool
+
+	if caPath != "" {
+		caCertPool, err := fetchCACert(caPath)
+		if err != nil {
+			b.logger.Fatal("Failed to load server CA certificate", zap.Error(err))
+			return nil
+		}
+		caPool = caCertPool
+	}
+	if certPath != "" {
+		myCert, err := tls.LoadX509KeyPair(certPath, keyPath)
+		if err != nil {
+			b.logger.Fatal("Failed to load client certificate", zap.Error(err))
+			return nil
+		}
+		cert = &myCert
+	}
+	// If we are given arguments to verify either server or client, configure TLS
+	if caPool != nil || cert != nil {
+		tlsConfig := &tls.Config{
+			ServerName:   host,
```

Comment:
Is it required to set ServerName on tls.Config?  Most of the examples don't set it.

Reply:
>But this should match the name of the server we are trying to connect to rather than the host we are trying to connect from.

[It takes `address` argument from the command line](https://github.com/temporalio/temporal/pull/812/files#diff-6e1d3cc0dc6cdcbe5ea5283ea7c5c4b9R105), but defaults to `localHostPort` for development scenarios if

Reply:
Sounds good - just saw localHostPort and got confused there.